wgengine/magicsock/peermtu.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build (darwin && !ios) || (linux && !android)
+
+package magicsock
+
+// Peer path MTU routines shared by platforms that implement it.
+
+// DontFragSetting returns true if at least one of the underlying sockets of
+// this connection is a UDP socket with the don't fragment bit set, otherwise it
+// returns false.
+func (c *Conn) DontFragSetting() bool {
+	return c.getDontFragment("udp4") || c.getDontFragment("udp6")
+}
+
+// ShouldPMTUD returns true if this client should try to enable peer MTU
+// discovery, false otherwise.
+func (c *Conn) ShouldPMTUD() bool {
+	if v, ok := debugEnablePMTUD().Get(); ok {
+		if debugPMTUD() {
+			c.logf("magicsock: peer path MTU discovery set via envknob to %v", v)
```

Comment:
you might want to ensure that all of the pmtu logs share a common phrase, such as consistently doing:

```
c.logf("magicsock: pmtu: ...")
```

Not directly relevant to your PR, but we should also change c.logf sometime to logger.WithPrefix then we can remove our need to manual prefix `magicsock: `

Reply:
Done! Agreed on the log prefix change.